src/LtiAssignmentsGradesService.php

```diff
@@ -62,12 +91,24 @@ public function findOrCreateLineitem(LtiLineitem $newLineItem)
         return new LtiLineitem($createdLineItems['body']);
     }
 
+    public function findOrCreateLineitem(LtiLineitem $newLineItem): LtiLineitem
+    {
+        $lineitem = $this->findLineItem($newLineItem);
+        if ($lineitem !== null) { return $lineitem; }
+
+        return $this->createLineitem($newLineItem);
+    }
+
     public function getGrades(LtiLineitem $lineitem = null)
     {
         if ($lineitem !== null) {
-            $lineitem = $this->findOrCreateLineitem($lineitem);
+            // Skip line item lookup/creation if we already know the Id
+            if (empty($lineitem->getId())) {
```

Comment:
Seems like we're checking this a bunch. It might be worth abstracting to a function. Up to you.

Reply:
it's definitely weird looking code, because the getId() of the lineitem is also the URL that is used to submit scores.